### ISSUE TITLE
Adding attachment on some expenses causes them to disappear from transactions list

### ISSUE THREAD

**Issue by pwsoftwaredesigns** [firefly-iii/firefly-iii#5694]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I was going through my transactions for the month of January adding receipts as attachments to each one.  On a few of them, after adding the attachment, the transactions disappeared from the list (Generated by "/transactions/withdrawal/2022-01-01/2022-01-31").  I ran a search to see what happened.   The transactions still exists, the date is correct, and the destination account is correct, however it just doesn't appear in the list.

It appears completely random as to which transactions do this.  In one instance (see screenshots) only half a split transaction

**TRANSACTIONS FOR JANUARY**
![Screenshot from 2022-02-02 08-02-35](https://user-images.githubusercontent.com/59979943/152169043-d36f6d9a-f5ab-456c-aa9d-e5b33cc4c738.png)

**SEARCH FOR "DENTIST"**
![Screenshot from 2022-02-02 08-00-55](https://user-images.githubusercontent.com/59979943/152169040-84e3ceed-f174-4f3f-ac2d-8e581d69f79f.png)

 disappears.  In another case, an entire, single transactions disappears.

### Expected behaviour

The transactions should not disappear

### Debug information

Debug information generated at 2022-02-02 08:06:23 America/Chicago for Firefly III version **5.5.13**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.5.13 |
| Firefly III API | 1.5.2 |
| PHP | 7.4.23 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Installation ID | a3ac0204-428a-4e2c-a13b-9b393d031272 |
| Using docker? | false |
| Telemetry | false |
| Layout | v1 |
| System TZ | America/Chicago |
| Browser TZ | America/Chicago |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | notice, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| Default language | en_US |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql |
| Current driver | mysql |
| DB version | 16 (exp. 16) |
| Login provider | eloquent  |
| Trusted proxies (.env) |  |

| User info | Value |
| --- | --- |
| Session start | 2022-02-01 00:00:00 |
| Session end | 2022-02-28 23:59:59 |
| Session first | 2020-12-31 00:00:00 |
| User ID | 1 |
| User language | en_US |
| User locale | en_US |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| User agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.71 Safari/537.36 |

**Comment by JC5** [comment]
I'll check it out, thanks for letting me know. Are these split transactions or normal ones?

**Comment by JC5** [comment]
Alright so they disappear from the list, but you can otherwise still see them when you visit them directly, or search for them.

What happens after you visit `/flush`? Does that help?

**Comment by pwsoftwaredesigns** [comment]
Yes, that’s correct. It hasn’t affected any of the totals and you can find them by searching directly, they just don’t appear sorted by the date.

I tried visiting /flush as you suggested, but that didn’t seem to fix the issue.

**Comment by JC5** [comment]
I've been playing with this for a while now but I haven't been able to replicate it unfortunately. Is there a javascript error in the console somewhere?

**Comment by pwsoftwaredesigns** [comment]
Sorry for not getting back to this sooner.  I took a look at the console when loading the transaction list for January (that should contain both splits) and do not see any Javascript errors.  If I click on the on part of the split that does appear in the list, it brings me to the transaction information page which does indeed show both parts of the split with the correct date (see my image).

I even went to /transactions/withdrawal/2022-01-22/2022-01-22?page=1 to see just the transactions for that date, and it still does not appear.

![image](https://user-images.githubusercontent.com/59979943/154277305-0acd8d62-d612-4125-b73a-3721fb63a814.png)

**Comment by JC5** [comment]
There's no attachment in your screenshot.

**Comment by pwsoftwaredesigns** [comment]
Okay, now I've discovered something odd.  I tried re-adding my attachment (a 20MB PDF file) and updating the transaction.  I did this and still no attachment showed.  So I added a different attachment to test (a 50kB JPEG file) and that one appeared.  And suddenly, that made the second half of the split appear correctly on the transactions list page.  Finally, I deleted my test attachment.  The transaction still shows on the list that it has an attachment, but when I view the transaction details, no attachment is listed (see my screenshots).

Is it possible the original attachment didn't upload completely because it is so large?
Is there a way I can "purge" all attachment data for the transaction?  Perhaps something got corrupted.

![image](https://user-images.githubusercontent.com/59979943/154492041-7f8e0126-a886-40ec-b9d3-481b3efae585.png)
![image](https://user-images.githubusercontent.com/59979943/154492088-73148eba-e749-4ff1-b930-493e67b95541.png)
